File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()) return false;
        int check=0;
        for(int i=0;i<s.size();i++)
        {
            check=0;
            
            for (int j=0;j<t.size();j++)
            {
                if (s[i] == t[j])
                {
                    s.erase(s.begin()+i);
                    t.erase(t.begin()+j);
                    check = 1;
                    i--;
                    break;
                }
            }
            if (check==0) return false;
        }
        if (s==t && s=="") return true;
        return false;
        
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_map = {}
        t_map = {}

        for i in s:
            if i not in s_map:
                s_map[i] = 1
            else:
                s_map[i] += 1

        for i in t:
            if i not in t_map:
                t_map[i] = 1
            else:
                t_map[i] += 1
        

        for i in s_map:
            if i in t_map and s_map[i] == t_map[i]:
                del t_map[i]
            else:
                return False
        
        if t_map == {}:
            return True

        return False
        